addons/sale/static/tests/sale_order_line_field.test.js

```diff
@@ -0,0 +1,266 @@
+    product_id = fields.Many2one({ relation: 'product' })
+    sequence = fields.Integer()
+    linked_line_id = fields.Many2one({ relation: 'sale.order.line' })
+    display_type = fields.Selection({
+        default: false,
+        selection: [
+            ['line_section', "Section"],
+            ['line_subsection', "Subsection"],
+            ['line_note', "Note"],
+        ],
+    })
+    product_type = fields.Selection({
+        default: 'product',
+        selection: [
+            ['product', "Product"],
+            ['combo', "Combo"],
+        ],
+    })
+}
+
+class SaleOrder extends models.Model {
+    _name = 'sale.order'
+    _records = [
+        {
+            id: 1,
+            name: "Combo Sale order",
+            order_line: SaleOrderLine._records.map(record => record.id),
+        },
+    ]
+
+    name = fields.Char()
+    order_line = fields.One2many({ relation: 'sale.order.line' })
+}
+
+defineComboModels();
+defineModels([SaleOrderLine, SaleOrder]);
+defineMailModels();
+
+test("test combo move up/down", async () => {
+    await mountView({
+        type: 'form',
+        resModel: 'sale.order',
+        resId: 1,
+        arch: `
+            <form>
+                <field
+                    name="order_line"
+                    widget="sol_o2m"
+                    options="{'subsections': True}"
+                >
+                    <list editable="bottom">
+                        <control>
+                            <create name="add_line_control" string="Add a line"/>
+                            <create name="add_section_control" string="Add a section" context="{'default_display_type': 'line_section'}"/>
+                            <create name="add_note_control" string="Add a note" context="{'default_display_type': 'line_note'}"/>
+                        </control>
+                        <field name="sequence" widget="handle" invisible="combo_item_id"/>
+                        <field name="name"/>
+                        <field name="display_type" column_invisible="1"/>
+                        <field name="linked_line_id" column_invisible="1"/>
+                        <field name="product_type" column_invisible="1"/>
+                        <field name="combo_item_id" column_invisible="1"/>
+                    </list>
+                </field>
+            </form>
+        `,
+    });
+
+    expect(queryAllTexts('.o_data_row')).toEqual([
+        "Non Combo Line1",
+        "Non Combo Line2",
+        "Test Combo1",
+        "Combo1 Item 1",
+        "Combo1 Item 2",
+        "Test Combo2",
+        "Combo2 Item 1",
+        "Combo2 Item 2",
+        "Non Combo Line3",
+    ]);
+
+    await contains('.o_data_row:contains(Test Combo1) .o_list_section_options button').click();
+    expect('.o-dropdown-item:contains(Move Up)').toHaveCount(1, {
+        message: 'Move up option should be there for Test combo1 line having SO lines before and after'
```

Comment:
```suggestion
        message: 'Move Up option should be there since Test combo1 has SO lines before and after'
```
Similarly you can change at other places to make it under 100